components/bootloader/Kconfig.projbuild

```diff
@@ -151,6 +151,32 @@ menu "Bootloader config"
             bool "Reset on GPIO high"
     endchoice
 
+    config BOOTLOADER_WAIT_AFTER_EN
+        bool "Enable Waiting time before Factory Reset GPIO probe (especially usefull for GPIO 0 as factory reset)"
+        default y
+        depends on BOOTLOADER_FACTORY_RESET
+        help
+            Enable Wait Time in bootloader prior to probing for Factory Reset condition.
+            This is especially usefull if GPIO 0 (BOOT button on most Dev Boards) is set as Factory Reset GPIO.
+            Without the wait time, the GPIO 0 / BOOT, if held down, during boot, is already probed by the ROM loader
+            which would be selecting UART boot mode.
+
+            The intended use case is as follows:
+               * set GPIO 0 (BOOT) as Factory Reset, which is already a pre-installed button on almost every ESP32 board
+               * for the GPIO 0 (BOOT): set 'Reset on GPIO Low'
+               * enable the Waiting Time after EN
+               * press RESET (EN) --> ROM Loader determines normal flash boot
+               * shortly after, press and hold GPIO 0 (BOOT) button until the Factory reset condition is recognized
+               * boot continues, Factory reset is applied
+
+    config BOOTLOADER_WAIT_AFTER_EN_TIME_MS
+        int "Time to wait before the Factory Reset GPIO probe (ms)"
+        depends on BOOTLOADER_WAIT_AFTER_EN
+        default 3000
```

Comment:
3 seconds sounds too much in my opinion as this would slow down the boot time by a lot

Reply:
1 s would be enough, you are right, it should be enough to release reset and pres
ss hold the boot button.. feel free to suggest a different value

Reply:
changed to 1 s, see current files state in merge request